lib/block-supports/layout.php

```diff
@@ -160,13 +160,15 @@ function gutenberg_render_layout_support_flag( $block_content, $block ) {
 		$used_layout = $default_layout;
 	}
 
-	$id    = uniqid();
-	$style = gutenberg_get_layout_style( ".wp-container-$id", $used_layout, $has_block_gap_support );
+	$id              = uniqid();
+	$style           = gutenberg_get_layout_style( ".wp-container-$id", $used_layout, $has_block_gap_support );
+	$container_class = 'wp-container-' . $id . ' ';
+	$justify_class   = $used_layout['justifyContent'] ? 'wp-justify-' . $used_layout['justifyContent'] . ' ' : '';
```

Comment:
Why we added this to the `layout` abstraction? Was something specific to pagination blocks?

Reply:
Yep, we needed to identify with a class the justification of the block, without this change that was not possible.

Reply:
There shouldn't be any block logic leaked into the `layout` abstraction. I'll check a bit the PR better to try to understand. That was the case recently with `Navigation` block as well, but we should find another way to handle specific block needs.

Reply:
Ok, I see, let me submit a PR with this change you propose.

Reply:
@ntsekouras a PR moving the functionality from Layout to Query Pagination code.
https://github.com/WordPress/gutenberg/pull/37113